lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/LocalSubmissionFromCS.java

```diff
@@ -0,0 +1,150 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.bridge.client;
+
+import org.apache.commons.lang.Validate;
+import org.apache.reef.client.parameters.DriverConfigurationProviders;
+import org.apache.reef.io.TcpPortConfigurationProvider;
+import org.apache.reef.runtime.common.files.REEFFileNames;
+import org.apache.reef.runtime.common.launch.parameters.DriverLaunchCommandPrefix;
+import org.apache.reef.runtime.local.client.LocalRuntimeConfiguration;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Configurations;
+import org.apache.reef.tang.Tang;
+import org.apache.reef.wake.remote.ports.parameters.TcpPortRangeBegin;
+import org.apache.reef.wake.remote.ports.parameters.TcpPortRangeCount;
+import org.apache.reef.wake.remote.ports.parameters.TcpPortRangeTryCount;
+
+import java.io.File;
+import java.util.ArrayList;
+
+/**
+ * Represents a job submission from the CS code.
+ * <p/>
+ * This class exists mostly to parse and validate the command line parameters provided by the C# class
+ * `Org.Apache.REEF.Client.Local.LocalClient`
+ */
+final class LocalSubmissionFromCS {
+  private final File driverFolder;
+  private final File jobFolder;
+  private final File runtimeRootFolder;
+  private final String jobId;
+  private final int numberOfEvaluators;
+  private final int tcpBeginPort;
+  private final int tcpRangeCount;
+  private final int tcpTryCount;
+
+  private LocalSubmissionFromCS(final File driverFolder,
+                                final String jobId,
+                                final int numberOfEvaluators,
+                                final int tcpBeginPort,
+                                final int tcpRangeCount,
+                                final int tcpTryCount) {
+    Validate.isTrue(driverFolder.exists(), "The driver folder does not exist.");
+    Validate.notEmpty(jobId, "The job is is null or empty.");
+    Validate.isTrue(numberOfEvaluators > 0, "The number of evaluators is <= 0.");
+    Validate.isTrue(tcpBeginPort >= 0, "The tcp start port given is < 0.");
+    Validate.isTrue(tcpRangeCount > 0, "The tcp range given is <= 0.");
+    Validate.isTrue(tcpTryCount > 0, "The tcp retry count given is <= 0.");
+    // We assume the given path to be the one of the driver. The job folder is one level up from there.
+    this.driverFolder = driverFolder;
+    this.jobFolder = driverFolder.getParentFile();
+    this.runtimeRootFolder = jobFolder.getParentFile();
```

Comment:
original folders are:
 final File jobFolder = new File(args[0]).getParentFile();
 final String runtimeRootFolder = jobFolder.getParentFile().getAbsolutePath();

Is the new runtimeRootFolder the same as previous one?